[cmake] Make sanitizers easy to use

neo1973 opened the pull request:
To use a sanitizer with clang or gcc add -DECM_ENABLE_SANITIZERS='san1;asn2;...'
as an option to your CMake command line. More documentation is provided in the
CMake file.

Note: This file is from the "Extra CMake Modules" package: https://api.kde.org/ecm/module/ECMEnableSanitizers.html

fetzerch commented:
Nice, that's definitively something I want to play with.
@hudokkow: What do you think about putting in external modules? It should probably go into the modules folder, should we have a subfolder for these?

hudokkow commented:
Fine by me. Name subfolder "extra"?

fetzerch commented:
Yes sounds good, @neo1973 mind moving it to modules/extra?

If you want, you could also mention it in https://github.com/xbmc/xbmc/blob/master/project/cmake/README.md. Maybe in a section before "Debugging the build".

neo1973 committed:
[cmake] Make sanitizers easy to use

To use a sanitizer with clang or gcc add -DECM_ENABLE_SANITIZERS='san1;san2;...'
as an option to your CMake command line. More documentation is provided in the
CMake file.

neo1973 committed:
Mention the Sanitizers in the README

neo1973 commented:
Done. I hope everything is OK now?

fetzerch commented:
Yes, looks good now. Thanks much!
